### PR TITLE
[WIP DO NOT MERGE] Address workaround for existing bug & include details for cloud provider

### PR DESCRIPTION
**Pull request by kerneltime**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1469)

### CODE REVIEW

File: docs/getting-started-guides/vsphere.md [hunk 1]
```diff
@@ -5,13 +5,47 @@ assignees:
...
+On a vSphere cluster it is recommended to use the vSphere cloud provider. vSphere cloud provider currently supports 
+
+1. [Volumes](http://kubernetes.io/docs/user-guide/volumes)
+2. [Persistent Volumes](http://kubernetes.io/docs/user-guide/persistent-volumes/)
+   1. [Static](http://kubernetes.io/docs/user-guide/persistent-volumes/#static)
+   2. [Dynamic](http://kubernetes.io/docs/user-guide/persistent-volumes/#dynamic)
+3. [Storage Class](http://kubernetes.io/docs/user-guide/persistent-volumes/#storageclasses)
+
+## Configuring vSphere Cloud Provider
+
```
Comment (pdhamdhere) [inline docs/getting-started-guides/vsphere.md]: Please add steps:
1. Create Cloud Configuration file on Master node. e.g. .etc/vsphere-cloud.conf
2. Document parameters in cloud config
3. Add command line demonstrating how to pass cloud-config to kube executables

I assume all of these steps are not needed if cluster is provisioned using kube-up

**Comment by jbeda** [comment]
A couple of quick questions/comments:
* It might be a good idea to separate out the documentation around the cloud provider from the `kube-up.sh` process.  Not everyone will want to use `kube-up.sh` and but may want to leverage the cloud provider.
* Has the underlying issue been fixed?  I'm confused why we don't just check this new fix in?  I'm happy to review that.
  * The fix looks like it might be isolated enough it would make sense to cherry-pick into the next 1.4 release.

**Comment by kerneltime** [comment]
@jbeda any recommendations for where we should document the cloud provider.. right now we are putting the usage guides into PV and PVC docs but the installation and configuring requirements for the cloud provider itself needs to be documented. Due to a lack of a better place we were thinking of putting it into the getting started page.

I am testing the cherry pick of the fixes for kube up (missing cloud provider config and salt stack bug) will post the PR for 1.4.<next> later today.

**Comment by jbeda** [comment]
The page you have here shows up under "Creating a Cluster" > "Running Kubernetes on Custom Solutions" > "On-Premise VMs" > "VMware vSphere" so it seems reasonable to put the documentation here.  Perhaps just break it out into a separate section?

I assume if the fix gets cherry picked into 1.4 you'd then edit this doc to remove the workaround?

**Comment by jbeda** [comment]
To be clear, if you document the cloud provider here, I'd move it after the `kube-up.sh` instructions and perhaps have a bit of an intro: 

"""
Kubernetes has adaptations for different environments called "Cloud Providers".  To enable extended support for vSphere in Kubernetes you can use the vSphere cloud provider....
"""

**Comment by kerneltime** [comment]
Thanks @jbeda taking a look at the documentation. If https://github.com/kubernetes/kubernetes/pull/34993 makes it into the next 1.4 release then we will not need the work around. 
Working on the documentation now.

**Pull request closed without merging**